apps/web/src/components/Basenames/RegistrationContext.tsx

```diff
@@ -202,11 +202,10 @@ export default function RegistrationProvider({ children }: RegistrationProviderP
       registrationStep === RegistrationSteps.Claim
     ) {
       logEventWithContext('register_name_transaction_processing', ActionType.change);
-
       setRegistrationStep(RegistrationSteps.Pending);
     }
 
-    if (transactionIsSuccess) {
+    if (transactionIsSuccess && registrationStep === RegistrationSteps.Pending) {
```

Comment:
are you sure you want this? I could see it introducing a race where this may not trigger if the transaction has succeeded and the step transitioned already